src/decode.c

```diff
@@ -569,6 +641,20 @@ void DecodeRegisterPerfCounters(DecodeThreadVars *dtv, ThreadVars *tv)
     dtv->counter_erspan = StatsRegisterMaxCounter("decoder.erspan", tv);
     dtv->counter_nsh = StatsRegisterMaxCounter("decoder.nsh", tv);
     dtv->counter_flow_memcap = StatsRegisterCounter("flow.memcap", tv);
+    /* Register stats counters for all valid exception policy values */
+    const char *eps_flow_str = "flow.memcap_exception_policy.";
+    bool is_eps_valid = false;
+    for (uint8_t i = 0; i < EXCEPTION_POLICY_MAX; i++) {
+        is_eps_valid = IsFlowMemcapExceptionPolicyStatsValid(i);
+        if (is_eps_valid) {
+            /* "flow.memcap_exception_policy." + longest exception policy string size
+             * ("pass_packet") = 41 */
+            char eps_flow_stats[41];
+            snprintf(eps_flow_stats, sizeof(eps_flow_stats), "%s%s", eps_flow_str,
+                    ExceptionPolicyEnumToString(i));
+            dtv->counter_flow_memcap_eps.eps_id[i] = StatsRegisterCounter(eps_flow_stats, tv);
```

Comment:
This triggers the `stack use after scope` error with Address Sanitizer. Haven't investigated enough.